docs/source/https.md

```diff
@@ -0,0 +1,53 @@
+# Enabling HTTPS
+
+In order to protect the data and passwords transfered over the public network, you should not run JupyterHub without enabling **HTTPS**.
+
+JupyterHub Traefik-Proxy supports **automatically** configuring HTTPS via [Let’s Encrypt](https://letsencrypt.org/docs/), or setting it
+up **manually** with your own key and certificate.
+
+## How-To HTTPS for JupyterHub managed Traefik-Proxy
+1. Via **Let'Encrypt**:
+    * Enable automatic https:
+    ```python
+    c.Proxy.traefik_auto_https=True
+    ```
+    * Configure a HTTPS port. This is where all the traffic coming through the HTTP entrypoint will be redirected to:
+    ```python
+    c.Proxy.traefik_https_port=443
+    ```
+    * Set the email address used for Let's Encrypt registration:
+    ```python
+    c.Proxy.traefik_letsencrypt_email="jovyan@jupyter.test"
+    ```
+    * Set the domain list:
+    ```python
+    c.Proxy.traefik_letsencrypt_domains=["jupyter.test"]
+    ```
+    * Set the the CA server to be used:
+    ```python
+    c.Proxy.traefik_acme_server="https://acme-v02.api.letsencrypt.org/directory"
+    ```
+    <span style="color:green">**Note !**</span>
+
+    **TraefikProxy**, supports only the most common challenge type, i.e. the [HTTP-01 ACME challenge](https://letsencrypt.org/docs/challenge-types/#http-01-challenge).
+    If other challenge type is needed, one could setup the proxy to be externally managed to get access to all the Traefik's configuration options (including the
+    ACME challenge type).
+
+2. **Manually**, by providing your own key and certificate:
+
+    Providing a certificate and key can be done by configuring JupyterHub to enable SSL encryption as specified in [the docs](https://jupyterhub.readthedocs.io/en/stable/getting-started/security-basics.html?highlight=https#enabling-ssl-encryption). Example:
+    ```python
+      c.JupyterHub.ssl_key = '/path/to/my.key'
+      c.JupyterHub.ssl_cert = '/path/to/my.cert'
+    ```
+
+    Traefik will then redirect all HTTP traffic to the HTTPS entrypoint. The default HTTPS entrypoint port is 8843, but can be configured through:
+    ```python
+    c.Proxy.traefik_https_port=443
+    ```
```

Comment:
It is fine to describe this configuration twice, just wanted to make you aware of that is done and you have the option to avoid writing it two times.

Reply:
I thought it would be a good idea to have it here also for the cases when users just come to this section from the table of contents and don't necessarily read what's in the section above because that isn't the use case they are looking for.